src/main/java/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -57,4 +57,10 @@ public void update(ShoppingCart shoppingCart) {
                     + shoppingCart.getId(), e);
         }
     }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.setTickets(null);
```

Comment:
```suggestion
        shoppingCart.setTickets(new ArrayList<>());
```
у твоєму варіанті, якщо очистити корзину, а потім додати новий квиток, може вилетіти NPE. перевір, якщо не вилітатиме, можеш не змінювати